src/pvrclient-nextpvr.cpp

```diff
@@ -514,7 +516,7 @@ PVR_ERROR cPVRClientNextPVR::GetEpg(ADDON_HANDLE handle, int iChannelUid, time_t
       XBMC->Log(LOG_DEBUG, "Skipping expired EPG data %d %ld %lld",iChannelUid,iStart, iEnd);
       return PVR_ERROR_INVALID_PARAMETERS;
   }
-  sprintf(request, "/service?method=channel.listings&channel_id=%d&start=%d&end=%d&genre=all", iChannelUid, (int)iStart, (int)iEnd);
+  sprintf(request, "/service?method=channel.listings&channel_id=%d&start=%ld&end=%ld&genre=all", iChannelUid, static_cast<long int> (iStart), static_cast<long int> (iEnd));
```

Comment:
For you cast just this: `static_cast<long int>(iStart)`. So no space between `>` anf `(`.

Why long int instead of int?

Reply:
OK I can fix the space.  I used long int because PVR defines the start and end time as time_t

Reply:
Then you are better off to use `%lld` and `long long`.

Reply:
That will be the same on all platforms.

Reply:
OK I made those changes, I don't expect the add-on will or the backend work past 2038 without change.  Thanks again I will force it through if there are not other comments.

Reply:
My point was that `int` and `long int`. are the same. Both guarantee at least 32 bits.

If you wanted to be correct which I assume you did `long long` is correct. Using `long int` makes no sense.

Reply:
Ok I will just use int like other pvr's thanks.